pandas/core/missing.py

```diff
@@ -621,3 +625,29 @@ def fill_zeros(result, x, y, name, fill):
             result = result.reshape(shape)
 
     return result
+
+
+def validate_fill_value(value, dtype):
+    if is_list_like(value) or is_dict_like(value) or callable(value):
```

Comment:
why dont' you check ``not is_scalar``? (which allows strings, datetimes, and all pandas scalars).